AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/Connection/DataStoreConnectionScenario6Tests.swift

```diff
@@ -90,6 +90,124 @@ class DataStoreConnectionScenario6Tests: SyncEngineIntegrationTestBase {
+            return
+        }
+
+        guard let fetchedPost = fetchedComment.post else {
+            XCTFail("Post is nil, should be loaded")
+            return
+        }
+
+        guard let fetchedBlog = fetchedPost.blog else {
+            XCTFail("Blog is nil, should be loaded")
+            return
+        }
+
+        XCTAssertEqual(fetchedPost.id, post.id)
+        XCTAssertEqual(fetchedPost.title, post.title)
+
+        XCTAssertEqual(fetchedBlog.id, blog.id)
+        XCTAssertEqual(fetchedBlog.name, blog.name)
+    }
+
+    func testGetPostThenFetchBlogAndComment() throws {
+        try startAmplifyAndWaitForSync()
+        guard let blog = saveBlog(name: "name"),
+              let post = savePost(title: "title", blog: blog),
+              let comment = saveComment(post: post, content: "content") else {
+            XCTFail("Could not create blog, post, and comment")
+            return
+        }
+
+        // fetch post
+        let getPostCompleted = expectation(description: "get post complete")
+        var resultPost: Post6?
+        Amplify.DataStore.query(Post6.self, byId: post.id) { result in
+            switch result {
+            case .success(let queriedPostOptional):
+                guard let queriedPost = queriedPostOptional else {
+                    XCTFail("Could not get post")
+                    return
+                }
+                XCTAssertEqual(queriedPost.id, post.id)
+                resultPost = queriedPost
+                getPostCompleted.fulfill()
+            case .failure(let response):
+                XCTFail("Failed with: \(response)")
+            }
+        }
+        wait(for: [getPostCompleted], timeout: TestCommonConstants.networkTimeout)
+
+        guard let fetchedPost = resultPost else {
+            XCTFail("Could not get post")
+            return
+        }
+
+        guard let eagerlyLoadedBlog = fetchedPost.blog else {
+            XCTFail("Blog is nil, should be loaded")
+            return
+        }
+
+        XCTAssertEqual(eagerlyLoadedBlog.id, blog.id)
+        XCTAssertEqual(eagerlyLoadedBlog.name, blog.name)
+        if let postsInEagerlyLoadedBlog = eagerlyLoadedBlog.posts {
+            XCTAssertEqual(postsInEagerlyLoadedBlog.count, 1)
+            XCTAssertTrue(postsInEagerlyLoadedBlog.contains(where: {(postIn) -> Bool in
+                postIn.id == post.id
+            }))
+            XCTAssertEqual(postsInEagerlyLoadedBlog[0].id, post.id)
+        }
+
+
+        guard let lazilyLoadedComments = fetchedPost.comments else {
+            XCTFail("Could not get comments")
+            return
+        }
+
+        guard case .notLoaded = lazilyLoadedComments.loadedState else {
+            XCTFail("Should not be in loaded state")
+            return
+        }
+        XCTAssertEqual(lazilyLoadedComments.count, 1)
+        XCTAssertTrue(lazilyLoadedComments.contains(where: { (commentIn) -> Bool in
+            commentIn.id == comment.id
+        }))
```

Comment:
if there's only one and we are sure of it from the XCTAssertEqual, then we can probably make due with 
```
XCTAssertEqual(lazilyLoadedComments.count, 1)
XCTAssertTrue(lazilyLoadedComments[0], comment.id)
```